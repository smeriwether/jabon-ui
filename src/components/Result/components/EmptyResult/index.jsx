import React from 'react';

import Logo from '../../../../UI/Logo';

const EmptyResult = () => {
  return (
    <div className='tc'>
      <Logo fill='gray' height='80' width='80' />
      <p className='gray f3 pt3'>
        There's nothing here, yet.
      </p>
    </div>
  );
};

export default EmptyResult;
