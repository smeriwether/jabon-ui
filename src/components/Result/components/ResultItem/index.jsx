import React from 'react';
import PropTypes from 'prop-types';

import Editor from '../../../Editor';
import { formatMilliseconds } from '../../../../utils/time';


const propTypes = {
  result: PropTypes.shape({
    id: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
    succeeded: PropTypes.bool.isRequired,
    runtime: PropTypes.number.isRequired,
    content: PropTypes.string
  }).isRequired
};

const ResultItem = ({ result }) => {
  const status = result.succeeded ? 'Success' : 'Failed';
  const runtime = formatMilliseconds(result.runtime);
  const dataRole = `result-item-${result.id}`

  return (
    <div className='h-100 w-100 pa3 bg-white ba b--moon-gray' data-role={ dataRole }>
      <div className='h-10 w-100 pb2 flex justify-between items-center'>
        <div className={ result.succeeded ? 'b dark-green' : 'b dark-red' } data-role='result-status'>
          { status }
        </div>
        <div className='silver'>
          { runtime }
        </div>
      </div>

      <div className='h-90 w-100 ba b--moon-gray' data-role='result-editor'>
        <Editor name={ 'code-editor-' + result.id } readOnly={ true } value={ result.content } />
      </div>
    </div>
  );
};

ResultItem.propTypes = propTypes;

export default ResultItem;
