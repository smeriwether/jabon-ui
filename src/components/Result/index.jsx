import React from 'react';
import PropTypes from 'prop-types';

import EmptyResult from './components/EmptyResult';
import ResultItem from './components/ResultItem';

const propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
    succeeded: PropTypes.bool.isRequired,
    runtime: PropTypes.number.isRequired,
    content: PropTypes.string
  }))
};

const Result = ({ results }) => {
  const empty = !results || results.length === 0;

  const emptyView = () => {
    return (
      <div className='h-100 w-100 flex justify-center items-center'>
        <EmptyResult />
      </div>
    );
  };

  const resultsView = () => {
    return results.map((result) => {
      return (
        <div className='h400 w-100 pb3' key={ result.id } >
          <ResultItem result={ result } />
        </div>
      );
    });
  };

  return (
    <div className='h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll'>
      { empty ? emptyView() : resultsView() }
    </div>
  );
};

Result.propTypes = propTypes;

export default Result;
