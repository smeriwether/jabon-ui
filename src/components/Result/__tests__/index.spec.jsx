import Result from '../../Result';

import EmptyResult from '../../Result/components/EmptyResult';
import ResultItem from '../../Result/components/ResultItem';

describe('<Result />', () => {
  const setup = (overrides) => {
    const props = {
      results: null,
      ...overrides
    };

    const wrapper = shallow(<Result { ...props } />);

    return { wrapper };
  };

  context('when there are no results', () => {
    it('should match the snapshot', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });

    it('should show the EmptyResult', () => {
      const { wrapper } = setup();
      expect(wrapper.find(EmptyResult).length).toEqual(1);
    });
  });

  context('when there are results', () => {
    it('should match the snapshot', () => {
      const results = [{ id: 1, succeeded: true, runtime: 123 }];
      const { wrapper } = setup({ results });
      expect(wrapper).toMatchSnapshot();
    });

    it('should show the ResultItem for each result', () => {
      const results = [
        { id: 1, succeeded: true, runtime: 123 }, { id: 2, succeeded: true, runtime: 123 }
      ];
      const { wrapper } = setup({ results });
      expect(wrapper.find(ResultItem).length).toEqual(2);
    });
  });
});
