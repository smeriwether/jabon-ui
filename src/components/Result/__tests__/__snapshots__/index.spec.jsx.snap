// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> when there are no results should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result
    results={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="h-100 w-100 flex justify-center items-center"
      >
        <EmptyResult />
      </div>,
      "className": "h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <EmptyResult />,
        "className": "h-100 w-100 flex justify-center items-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="h-100 w-100 flex justify-center items-center"
        >
          <EmptyResult />
        </div>,
        "className": "h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EmptyResult />,
          "className": "h-100 w-100 flex justify-center items-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Result /> when there are results should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result
    results={
      Array [
        Object {
          "id": 1,
          "runtime": 123,
          "succeeded": true,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h400 w-100 pb3"
        >
          <ResultItem
            result={
              Object {
                "id": 1,
                "runtime": 123,
                "succeeded": true,
              }
            }
          />
        </div>,
      ],
      "className": "h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ResultItem
            result={
              Object {
                "id": 1,
                "runtime": 123,
                "succeeded": true,
              }
            }
          />,
          "className": "h400 w-100 pb3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "result": Object {
              "id": 1,
              "runtime": 123,
              "succeeded": true,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h400 w-100 pb3"
          >
            <ResultItem
              result={
                Object {
                  "id": 1,
                  "runtime": 123,
                  "succeeded": true,
                }
              }
            />
          </div>,
        ],
        "className": "h-100 w-100 pa3 bg-light-gray bb br b--moon-gray overflow-scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ResultItem
              result={
                Object {
                  "id": 1,
                  "runtime": 123,
                  "succeeded": true,
                }
              }
            />,
            "className": "h400 w-100 pb3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "result": Object {
                "id": 1,
                "runtime": 123,
                "succeeded": true,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
